core/src/main/java/com/linecorp/armeria/common/logging/RequestLogListenerInvoker.java

```diff
@@ -34,12 +36,38 @@
      * Invokes {@link RequestLogListener#onRequestLog(RequestLog)}.
      */
     public static void invokeOnRequestLog(RequestLogListener listener, RequestLog log) {
+        requireNonNull(listener, "listener");
+        requireNonNull(log, "log");
         try (SafeCloseable ignored = log.context().push()) {
             listener.onRequestLog(log);
         } catch (Throwable e) {
             logger.warn("onRequestLog() failed with an exception:", e);
         }
     }
 
+    /**
+     * Invokes {@link RequestLogListener#onRequestLog(RequestLog)}.
+     */
+    static void invokeOnRequestLog(RequestLogListener[] listeners, RequestLog log) {
+        requireNonNull(listeners, "listeners");
+        requireNonNull(log, "log");
+        if (listeners.length == 0) {
+            return;
+        }
+
+        try (SafeCloseable ignored = log.context().push()) {
+            for (RequestLogListener listener : listeners) {
+                if (listener == null) {
+                    continue;
```

Comment:
Shouldn’t this be `break`?

Reply:
I thought either way is fine because the listener can be `null`. But let me change that. 😄

Reply:
If `listener` cannot be `null`, you could remove the `if` completely.

Reply:
But it still seems to me that `listener` can be `null`, according to `removeSatisfiedListeners()`.

Reply:
Ah yes, didn't read code carefully. 😅